commission/wizards/commission_make_settle.py

```diff
@@ -66,26 +67,28 @@ def _get_next_period_date(self, agent, current_date):
         elif agent.settlement == "annual":
             return current_date + relativedelta(years=1)
 
-    def _get_settlement(self, agent, company, sett_from, sett_to):
+    def _get_settlement(self, agent, company, currency, sett_from, sett_to):
         self.ensure_one()
         return self.env["commission.settlement"].search(
             [
                 ("agent_id", "=", agent.id),
                 ("date_from", "=", sett_from),
                 ("date_to", "=", sett_to),
                 ("company_id", "=", company.id),
+                ("currency_id", "=", currency.id),
                 ("state", "=", "settled"),
                 ("settlement_type", "=", self.settlement_type),
             ],
             limit=1,
         )
 
-    def _prepare_settlement_vals(self, agent, company, sett_from, sett_to):
+    def _prepare_settlement_vals(self, agent, company, currency, sett_from, sett_to):
```

Comment:
Changing a signature in a stable version is not correct, as it will convert in incompatible all the existing overwrites of this method. Don't you have any other way?